src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -414,61 +415,119 @@ public bool TryCreateType(ITypeSymbol symbol, [NotNullWhen(true)] out CSharpType
+            Type? existingType = TryGetFrameworkType(namedTypeSymbol);
 
-            if (existingType != null && validator(existingType))
+            if (existingType is not null && validator(existingType))
             {
-                var arguments = namedTypeSymbol.TypeArguments.Select(a => CreateType(a)).ToArray();
+                if (!TryPopulateArguments(namedTypeSymbol.TypeArguments, validator, out var arguments))
+                {
+                    return false;
+                }
                 type = new CSharpType(existingType, false, arguments);
             }
             else
             {
                 type = _library.FindTypeByName(namedTypeSymbol.Name);
             }
 
-            if (type == null)
+            if (type is null)
             {
                 return false;
             }
 
             if (!type.IsValueType &&
-                symbol.NullableAnnotation != NullableAnnotation.NotAnnotated)
+                namedTypeSymbol.NullableAnnotation != NullableAnnotation.NotAnnotated)
             {
                 type = type.WithNullable(true);
             }
 
             return true;
         }
 
+        private bool TryCreateTypeForIArrayTypeSymbol(IArrayTypeSymbol symbol, Func<Type, bool> validator, [NotNullWhen(true)] out CSharpType? type)
+        {
+            type = null;
+            if (symbol is not IArrayTypeSymbol arrayTypeSymbol)
+            {
+                return false;
+            }
+
+            // For IArrayTypeSymbol, we can only handle it when the element type is a framework type.
+            var arrayType = TryGetFrameworkType(arrayTypeSymbol);
+            if (arrayType is not null && validator(arrayType))
+            {
+                type = new CSharpType(arrayType, arrayType.IsValueType && symbol.NullableAnnotation != NullableAnnotation.NotAnnotated);
+                return true;
+            }
+            return false;
+        }
+
+        private Type? TryGetFrameworkType(ISymbol namedTypeSymbol)
+        {
+            var fullMetadataName = GetFullMetadataName(namedTypeSymbol);
+            var fullyQualifiedName = $"{fullMetadataName}, {namedTypeSymbol.ContainingAssembly?.Name}";
+            return Type.GetType(fullMetadataName) ?? Type.GetType(fullyQualifiedName);
+        }
+
+        // There can be argument type missing
+        private bool TryPopulateArguments(ImmutableArray<ITypeSymbol> typeArguments, Func<Type, bool> validator, [NotNullWhen(true)] out CSharpType[]? arguments)
+        {
+            arguments = null;
+            var result = new List<CSharpType>();
+            foreach (var typeArgtment in typeArguments)
+            {
+                if (!TryCreateType(typeArgtment, validator, out CSharpType? type))
+                {
+                    return false;
+                }
+                result.Add(type);
+            }
+            arguments = result.ToArray();
```

Comment:
since [this PR](https://github.com/Azure/autorest.csharp/pull/4004) has been merged, this `ToArray` could be removed.